ramiro_serra/dif_driver_investigation.md

```diff
@@ -0,0 +1,53 @@
+
+Based on these observations, the configuration transition equation is
+
+![\begin{displaymath}\begin{split}{\dot x}& = \frac{r}{2} (u_l + u_r) \cos \theta ...
+...theta  {\dot \theta}& = \frac{r}{L} (u_r - u_l) . \end{split}\end{displaymath}](http://planning.cs.uiuc.edu/img5526.gif)
+
+The translational part contains ![ \cos \theta](http://planning.cs.uiuc.edu/img1449.gif) and ![ \sin \theta](http://planning.cs.uiuc.edu/img1450.gif) parts, just like the simple car because the differential drive moves in the direction that its drive wheels are pointing. 
+
+The translation speed depends on the average of the angular wheel velocities. To see this, consider the case in which one wheel is fixed and the other rotates. This initially causes the robot to translate at '1/2' of the speed in comparison to both wheels rotating. The rotational speed ![ {\dot \theta}](http://planning.cs.uiuc.edu/img5488.gif) is proportional to the change in angular wheel speeds. The robot's rotation rate grows linearly with the wheel radius but reduces linearly with respect to the distance between the wheels
+
+## How to control it?
+
+#### The unicycle aproach
+
+Dealing with the displacement and velocities of the two wheels of a differential drive robot is messy. A preferred model is that of a unicycle, where we can think of the robot as having one wheel that can move with a desired velocity (V) at a specified heading Phi. Having the equations to translate between the unicycle model and our wheel velocities is what allows us to simplify the robot with the unicycle model. We have seen how to take measured wheel displacements to calculate the new robot pose. Now we do the reverse and calculate the desired wheel velocities from the unicycle model.
```

Comment:
What does ROS have in terms of velocity messages for the unicycle model?

Reply:
In case you addressed this kind of changes in another place, let me know because Github doesn't delete them.